curriculum/challenges/english/25-front-end-development/lab-movie-review-page/67ef8f695db52583424bd118.md

```diff
@@ -0,0 +1,202 @@
+demoType: onClick
+---
+
+# --description--
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
+
+**User Stories:**
+
+1. Your should have a `main` element.
+2. Inside the `main` element, you should have an `h1` element for the movie title.
+3. Below the `h1` element, you should have an `img` element displaying the movie cover, you can use `https://cdn.freecodecamp.org/curriculum/labs/rise-beyond-2.png` if you would like, and a descriptive `alt` attribute describing the image.
+4. You should have a `p` element containing a brief movie description.
+5. You should have another `p` element to display the movie rating. It should contain a `strong` element with the text `Movie Rating`.
+6. You should have a `span` element with an `aria-hidden` attribute set to `true` containing a visual representation of the rating using stars `⭐⭐⭐⭐⭐⭐⭐⭐⭐☆` and a numerical value in parentheses (e.g. `9.2/10`).  
+7. You should have a `h2` element with the text `Cast Members`.
+8. You should have a `ul` element.
+9. Inside the `ul` element, you should have multiple `li` elements each containing an actor's name wrapped in a `strong` element and their corresponding character role (e.g., `James Holloway as Ethan Carter`).  
+
+# --hints--
+
+You should have a `main` element.
+
+```js
+assert.isNotNull(document.querySelector("main"));
+```
+
+Inside the `main` element, you should have an `h1` element containing the movie title.
+
+```js
+const h1 = document.querySelector("main > h1");
+assert.isAbove(h1?.innerText.length, 0);
+```
+
+Below the `h1` element, you should have an `img` element with a descriptive `alt` attribute.
+  
+```js
+const img = document.querySelector("main > h1 + img");
+assert.isNotNull(img);
+assert.isNotEmpty(img.getAttribute("alt"));
+```
+
+You should have a `p` element after the image that contains the movie description.
+
+```js
+const description = document.querySelector("main img + p");
+assert.isNotNull(description);
+assert.isNotEmpty(description.innerText);
+```
+
+You should have another `p` element that contains the rating.
+
+```js
+const paragraphs = document.querySelectorAll("main p");
+assert.isAtLeast(paragraphs.length, 2);
+```
+
+Inside the second `p` element, you should have a `strong` element with the text `Movie Rating`.
+
+```js
+const strongEl = document.querySelector("main p:nth-of-type(2) strong");
+assert.match(strongEl?.innerText.trim(), /^Movie Rating:?$/);
+```
+
+There should be a `span` element inside the rating paragraph.
+
+```js
+const spanEl = document.querySelector("main p:nth-of-type(2) span");
+assert.isNotNull(spanEl);
+```
+
+The `span` element inside the rating paragraph should have ten stars, either filled in (⭐) or empty (☆), followed by a numerical value in parentheses.
+
+```js  
+const spanEl = document.querySelector("main p:nth-of-type(2) span");  
+assert.match(spanEl?.innerText, /^(?:[⭐☆]{10})\s*\(\s*\d+\.\d+\s*\/\s*10\s*\)$/);
```

Comment:
this doesn't account for whole integers. `<span>⭐⭐⭐⭐⭐⭐⭐⭐⭐☆(6/10)</span>` fails the test. this update should work both floats and ints

```suggestion
assert.match(spanEl?.innerText, /^(?:[⭐☆]{10})\s*\(\s*\d+(?:\.\d+)?\s*\/\s*10\s*\)$/);
```